conjure-go-client/httpclient/internal/request_retrier.go

```diff
@@ -111,24 +113,27 @@ func (r *RequestRetrier) getRetryFn(resp *http.Response, respErr error) func() {
 		// Immediately backoff and select the next URI.
 		// TODO(whickman): use the retry-after header once #81 is resolved
 		return r.nextURIAndBackoff
-	} else if isUnavailableResponse(resp, respErr) || resp == nil {
```

Comment:
We move `resp == nil` as the last condition, given for 307/308s the resp will be nil as well.